multigeiger/speaker.cpp

```diff
@@ -183,27 +182,33 @@ void setup_speaker(bool playSound, bool _led_tick, bool _speaker_tick) {
 
   setup_audio_timer(isr_audio, PERIOD_DURATION_US);
 
-  if (playSound) {
-    static int sequence[][4] = {
-      TONE(1174659, 1, -1, 2),  // D
-      TONE(0, 0, 0, 2),         // ---
-      TONE(1318510, 1, -1, 2),  // E
-      TONE(0, 0, 0, 2),         // ---
-      TONE(1479978, 1, -1, 2),  // Fis
-      TONE(0, 0, 0, 2),         // ---
-      TONE(1567982, 1, -1, 4),  // G
-      TONE(1174659, 1, -1, 2),  // D
-      TONE(1318510, 1, -1, 2),  // E
-      TONE(1174659, 1, -1, 4),  // D
-      TONE(987767, 1, -1, 2),   // H
-      TONE(1046502, 1, -1, 2),  // C
-      TONE(987767, 1, -1, 4),   // H
-      TONE(987767, 0, -1, 4),   // H
-      TONE(0, 0, 0, 2),         // ---
-      TONE(0, 0, -1, 0),        // speaker off, end
-    };
-    play((int *)sequence);
-  }
+  tick_enable(false);  // no ticking while we play melody / init sound
+
+  static int init[][4] = {
+    TONE(0, 0, 0, 0),        // speaker off, led off, end
+  };
+  play((int *)init);
+
+  static int melody[][4] = {
+    TONE(1174659, 1, -1, 2),  // D
+    TONE(0, 0, -1, 2),        // ---
+    TONE(1318510, 1, -1, 2),  // E
+    TONE(0, 0, -1, 2),        // ---
+    TONE(1479978, 1, -1, 2),  // Fis
+    TONE(0, 0, -1, 2),        // ---
+    TONE(1567982, 1, -1, 4),  // G
+    TONE(1174659, 1, -1, 2),  // D
+    TONE(1318510, 1, -1, 2),  // E
+    TONE(1174659, 1, -1, 4),  // D
+    TONE(987767, 1, -1, 2),   // H
+    TONE(1046502, 1, -1, 2),  // C
+    TONE(987767, 1, -1, 4),   // H
+    TONE(987767, 0, -1, 4),   // H
+    TONE(0, 0, -1, 2),        // ---
+    TONE(0, 0, -1, 0),        // speaker off, end
+  };
+  if (playSound)
+    play((int *)melody);
```

Comment:
I just changed the 3rd element of all TONE tuples to -1 "don't touch" - resetting the LED to 0 in the tone pauses is neither wanted nor needed.

Diff looks like more than it really is.